internal/internal_workflow_client_test.go

```diff
@@ -1239,7 +1285,221 @@ func (s *workflowClientTestSuite) SignalWithStartWorkflowWithMemoAndSearchAttr()
+	s.NoError(err)
+}
+
+func (s *workflowClientTestSuite) TestSignalWithStartWorkflow_RequestCreationFails() {
+	options := StartWorkflowOptions{
+		ID:                              workflowID,
+		TaskList:                        "", // this causes error
+		ExecutionStartToCloseTimeout:    timeoutInSeconds,
+		DecisionTaskStartToCloseTimeout: timeoutInSeconds,
+	}
+	wf := func(ctx Context) string {
+		return "result"
+	}
+
+	_, err := s.client.SignalWithStartWorkflow(context.Background(), "wid", "signal", "value", options, wf)
+	s.Error(err)
+}
+
+func (s *workflowClientTestSuite) TestSignalWithStartWorkflowAsync_RequestCreationFails() {
+	options := StartWorkflowOptions{
+		ID:                              workflowID,
+		TaskList:                        "", // this causes error
+		ExecutionStartToCloseTimeout:    timeoutInSeconds,
+		DecisionTaskStartToCloseTimeout: timeoutInSeconds,
+	}
+	wf := func(ctx Context) string {
+		return "result"
+	}
+
+	_, err := s.client.SignalWithStartWorkflowAsync(context.Background(), "wid", "signal", "value", options, wf)
+	s.Error(err)
+}
+
+func (s *workflowClientTestSuite) TestSignalWithStartWorkflowAsync_Error() {
+	options := StartWorkflowOptions{
+		ID:                              workflowID,
+		TaskList:                        tasklist,
+		ExecutionStartToCloseTimeout:    timeoutInSeconds,
+		DecisionTaskStartToCloseTimeout: timeoutInSeconds,
+	}
+	wf := func(ctx Context) string {
+		return "result"
+	}
+
+	s.service.EXPECT().SignalWithStartWorkflowExecutionAsync(gomock.Any(), gomock.Any(), gomock.Any(),
+		gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, errors.New("failed")).AnyTimes()
+
+	// Pass a context with a deadline so error retry doesn't take forever
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	_, err := s.client.SignalWithStartWorkflowAsync(ctx, "wid", "signal", "value", options, wf)
+	s.Error(err)
+}
+
+func (s *workflowClientTestSuite) TestStartWorkflowAsync() {
+	client, ok := s.client.(*workflowClient)
+	s.True(ok)
+	options := StartWorkflowOptions{
+		ID:                              workflowID,
+		TaskList:                        tasklist,
+		ExecutionStartToCloseTimeout:    timeoutInSeconds,
+		DecisionTaskStartToCloseTimeout: timeoutInSeconds,
+	}
+	f1 := func(ctx Context, r []byte) string {
+		return "result"
+	}
+
+	s.service.EXPECT().StartWorkflowExecutionAsync(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, nil)
+
+	_, err := client.StartWorkflowAsync(context.Background(), options, f1, []byte("test"))
+	s.Equal(getDefaultDataConverter(), client.dataConverter)
+	s.NoError(err)
+}
+
+func (s *workflowClientTestSuite) TestStartWorkflowAsync_WithDataConverter() {
+	dc := newTestDataConverter()
+	s.client = NewClient(s.service, domain, &ClientOptions{DataConverter: dc})
+	client, ok := s.client.(*workflowClient)
+	s.True(ok)
```

Comment:
yep. followed the pattern in other tests without too much thinking on best way to rewrite these tests. addressing this unnecessary assertion since it's a small/local change